fingerprint2.js

```diff
@@ -51,7 +51,27 @@
       }
       return target
     },
-    get: function (done) {
+    getOrDefaultPromise: function(defaultResult, defaultComponents){
+      var that = this;
+      return new Promise(function(resolve, reject) {
+        that.getPromise()
+          .then(resolve)
+          .catch(function(e) {
+            resolve([defaultResult, defaultComponents, e])
+          });
+      })
+    },
+    getPromise: function() {
+      var that = this;
```

Comment:
Please run standard (https://standardjs.com/) over the code. Sorry if this isn't documented yet.

Reply:
Are you referring to the semicolon? I am disappointed that not typing the semicolon made it into some sort of standard now :( But that is for another discussion :).
Will remove it of course.

Reply:
Yeah and a few other stylistic things in your patch as well. Just use the standard autoformatter :)

Reply:
How do I run the proper linter? I have tried to run 'gulp lint', but that gave me like bazillion errors due to the not used semicolon.